ast/compile.go

```diff
@@ -1305,6 +1312,44 @@ func (c *Compiler) checkDuplicateImports() {
 	}
 }
 
+func (c *Compiler) checkKeywordOverrides() {
+	for _, name := range c.sorted {
+		mod := c.Modules[name]
+		errs := checkKeywordOverrides(mod, c.strict)
+		for _, err := range errs {
+			c.err(err)
+		}
+	}
+}
+
+func checkKeywordOverrides(node interface{}, strict bool) Errors {
+	if !strict {
+		return nil
+	}
+
+	errors := Errors{}
+
+	WalkRules(node, func(rule *Rule) bool {
+		name := rule.Head.Name.String()
+		if _, override := strictKeywords[name]; override {
+			errors = append(errors, NewError(CompileErr, rule.Location, "rule overrides %s", name))
+		}
+		return false
```

Comment:
I usually have to look this up... But doesn't `true` mean "don't keep going down this branch"? I think we can return true here, then, since rules can't be nested...? 🤔

Reply:
Huh, I think you're right. If there's never any point in searching deeper, couldn't we just remove the boolean return entirely from the passed function signature?